pkg/auth/authenticator/password/ldappassword/url.go

```diff
@@ -0,0 +1,146 @@
+package ldappassword
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+
+	"github.com/go-ldap/ldap"
+)
+
+type Scheme string
+
+const (
+	SchemeLDAP  Scheme = "ldap"
+	SchemeLDAPS Scheme = "ldaps"
+)
+
+// LDAPURL holds a parsed RFC 2255 URL
+type LDAPURL struct {
+	// Scheme is ldap or ldaps
+	Scheme Scheme
+	// Host is the host:port of the LDAP server
+	Host string
+	// The DN of the branch of the directory where all searches should start from
+	BaseDN string
+	// The attribute to search for
+	QueryAttribute string
+	// The scope of the search. Can be ldap.ScopeWholeSubtree, ldap.ScopeSingleLevel, or ldap.ScopeBaseObject
+	Scope int
```

Comment:
If this were private or in a private struct, I'd have no issue.  Having a non-enumerated enum in a public field of a public type worries me.  Maybe I haven't spent enough time in Go to just be happy....